### PR TITLE
release.groovy: make it to use ec2 workers and consistent w/ Jenkinsfile

### PR DESCRIPTION
**Pull request by Quentin-M**

### CODE REVIEW

File: release.groovy [hunk 1]
```diff
@@ -1,10 +1,30 @@
-// Update https://jenkins-tectonic.prod.coreos.systems/job/tectonic-release
+/* Tips
+1. Keep stages focused on producing one artifact or achieving one goal. This makes stages easier to parallelize or re-structure later.
+1. Stages should simply invoke a make target or a self-contained script. Do not write testing logic in this Jenkinsfile.
+3. CoreOS does not ship with `make`, so Docker builds still have to use small scripts.
```
Comment (amrutac) [inline release.groovy]: Can you add the link to the release job in here?

File: release.groovy [hunk 2]
```diff
@@ -17,51 +37,24 @@ pipeline {
...
-              }
+        node('worker && ec2') {
+          withCredentials(creds) {
+            withDockerContainer(builder_image) {
+              checkout([$class: 'GitSCM', branches: [[name: "refs/tags/${params.releaseTag}"]], userRemoteConfigs: [[credentialsId: 'github-coreosbot', url: 'https://github.com/coreos/tectonic-installer.git']]])
+              sh """#!/bin/bash -ex
+                sh "mkdir -p \$(dirname $GO_PROJECT) && ln -sf $WORKSPACE $GO_PROJECT"
+                sh "go get github.com/golang/lint/golint"
+                sh """#!/bin/bash -ex
+                export PRE_RELEASE=${params.preRelease}
```
Comment (amrutac) [inline release.groovy]: The old script was also export aws credentials, is that not needed anymore?

File: release.groovy (hunk 1 above)
Comment (Quentin-M) [inline release.groovy]: Just did! Thanks~

**Comment by amrutac** [comment]
Any other PR comments for suggestions, modifications to this can be handled in a separate PR

**Pull request merged**